frontend/public/components/secrets/create-secret.jsx

```diff
@@ -0,0 +1,163 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { k8sCreate, k8sUpdate } from '../../module/k8s';
+import { ButtonBar, Firehose, history, kindObj, StatusBox } from '../utils';
+import { getActiveNamespace, formatNamespacedRouteForResource, UIActions } from '../../ui/ui-actions';
+import { SafetyFirst } from '../safety-first';
+import { WebHookSecretSubform } from './subforms';
+
+const determineSecretTypeAbstraction = (secret) => {
+  const { type, data } = secret;
+  switch (type) {
+    case 'kubernetes.io/dockerconfigjson':
+    case 'kubernetes.io/dockercfg':
+      return 'image';
+    case 'kubernetes.io/basic-auth':
+    case 'kubernetes.io/ssh-auth':
+      return 'source';
+    case 'Opaque':
+    case 'kubernetes.io/tls':
+    case 'kubernetes.io/service-account-token':
+      if (_.includes(_.keys(data), 'WebhookSecretKey')) {
+        return 'webhook';
+      }
+      return 'generic';
+    default:
+      return 'generic';
+  }
+};
+
+const BaseEditSecret = connect(null, {setActiveNamespace: UIActions.setActiveNamespace})(
+  class BaseEditSecret_ extends SafetyFirst {
+    constructor (props) {
+      super(props);
+      const kind = 'Secret';
+      const existingData = _.pick(props.obj, ['metadata.name', 'metadata.namespace', 'data']);
```

Comment:
agreed, I've picked those two mainly because of the `duplicate` action but since we've nuked that options I will put the whole `metadata` field there.